tests/shellcheck/shellcheck.bzl

```
@@ -1,16 +1,18 @@
 load("//tests:inline_tests.bzl", "sh_inline_test")
+load("@bazel_skylib//lib:shell.bzl", "shell")
 
-def shellcheck(name, args, data, sh_flavor = "sh", excludes = [], visibility = None):
-    excludes_arg = "--exclude=" + ",".join(excludes) if excludes else ""
-    shell_arg = "--shell " + sh_flavor if sh_flavor else ""
+def shellcheck(name, args, data, sh_flavor = "sh", visibility = None):
+    cmd = ["shellcheck", "--color=always"]
+    if sh_flavor != None:
+        cmd += ["--shell", sh_flavor]
     sh_inline_test(
         name = name,
         visibility = visibility,
         size = "small",
         args = args,
         data = data,
         script = """\
-shellcheck --color=always {} {} "$1"
-""".format(shell_arg, excludes_arg),
+{} "$1"
```

```suggestion
shellcheck "$@"
```
`args` is a list and could hold multiple values, correct? It just happens to only be called with a single argument at the time, so this never caused any issues.

Also, why not pass `--color=always` and optionally `--shell ...` in the `args` attribute to `sh_inline_test`? Then you don't have to worry about quoting.

---

Indeed, I've implemented your suggestion in dcac0a8. I believe the code has the simplest form possible now :+1: